src/common/TrackedOp.cc

```diff
@@ -52,17 +50,11 @@ void OpHistory::cleanup(utime_t now)
   while (arrived.size() &&
 	 (now - arrived.begin()->first >
 	  (double)(history_duration))) {
-    duration.erase(make_pair(
-	arrived.begin()->second->get_duration(),
-	arrived.begin()->second));
-    arrived.erase(arrived.begin());
```

Comment:
Here we're trimming based on the earliest arrival time, which is the time limited part of the equation. I think this is correct and makes sense?